packages/experimental/src/inbox-note/inbox-note.tsx

```diff
@@ -0,0 +1,300 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState, useRef } from '@wordpress/element';
+import { Button, Dropdown, Popover } from '@wordpress/components';
+import VisibilitySensor from 'react-visibility-sensor';
+import moment from 'moment';
+import classnames from 'classnames';
+import { H, Section } from '@woocommerce/components';
+import { sanitize } from 'dompurify';
+
+/**
+ * Internal dependencies
+ */
+import InboxNoteActionButton from './action';
+
+const ALLOWED_TAGS = [ 'a', 'b', 'em', 'i', 'strong', 'p', 'br' ];
+const ALLOWED_ATTR = [ 'target', 'href', 'rel', 'name', 'download' ];
+
+const sanitizeHTML = ( html: string ) => {
+	return {
+		__html: sanitize( html, { ALLOWED_TAGS, ALLOWED_ATTR } ),
+	};
+};
+
+type InboxNoteAction = {
+	id: number;
+	url: string;
+	label: string;
+	primary: boolean;
+	actioned_text?: boolean;
+};
+
+type InboxNote = {
+	id: number;
+	status: string;
+	title: string;
+	name: string;
+	content: string;
+	date_created: string;
+	date_created_gmt: string;
+	actions: InboxNoteAction[];
+	layout: string;
+	image: string;
+	is_deleted: boolean;
+	type: string;
+};
+
+type InboxNoteProps = {
+	note: InboxNote;
+	lastRead: number;
+	onDismiss?: ( note: InboxNote, type: 'all' | 'note' ) => void;
+	onNoteActionClick?: ( note: InboxNote, action: InboxNoteAction ) => void;
+	onBodyLinkClick?: ( note: InboxNote, link: string ) => void;
+	onNoteVisible?: ( note: InboxNote ) => void;
+};
+
+const DropdownWithPopoverProps = Dropdown as React.ComponentType<
+	Dropdown.Props & { popoverProps: Omit< Popover.Props, 'children' > }
+>;
+
+const InboxNoteCard: React.FC< InboxNoteProps > = ( {
+	note,
+	lastRead,
+	onDismiss,
+	onNoteActionClick,
+	onBodyLinkClick,
+	onNoteVisible,
+} ) => {
+	const [ clickedActionText, setClickedActionText ] = useState( false );
+	const hasBeenSeen = useRef( false );
+	const toggleButtonRef = useRef< HTMLButtonElement >( null );
+
+	const handleBodyClick = ( event: {
+		target: EventTarget | HTMLBaseElement;
```

Comment:
I don't think this type is correct? I think you can just type the whole arg as: `React.MouseEvent< HTMLElement >`. and I *think* the target is just typed as `EventTarget`?